torch/_library/custom_ops.py

```diff
@@ -502,6 +517,43 @@ def adinplaceorview_impl(keyset, *args, **kwargs):
                 with_keyset=True,
             )
 
+        from torch._library.utils import get_device_arg_id, has_tensor_arg
+
+        device_arg_id = None
+        if not has_tensor_arg(schema) and backend_select:
+            device_arg_id = get_device_arg_id(schema)
+            if device_arg_id is None:
+                raise ValueError(
+                    "Functions without tensor inputs are required to have a `device: torch.device` argument"
+                )
+
+        if device_arg_id is not None:
+
+            def get_dispatch_key(device: str):
```

Comment:
@zou3519 I'm wondering if there's a better way to get the dispatch key from a device string?

Reply:
try this: https://github.com/pytorch/pytorch/blob/64a04d2225800c137c1db562877ab43cd5d76f66/torch/_library/custom_ops.py#L266